Added JSON serialization format in logging exporter

tie opened the pull request:
**Description:**

This PR adds a JSON serializer for logs, traces and metrics, and an option to use it in the logging exporter. This makes it easier to `grep` debug logging output (i.e. filter lines containing specific string).

**Link to tracking Issue:** n/a

**Testing:** Added tests for validation JSON serialization output using testdata package.

**Documentation:** Added the description for new `format` option to the exporter’s README file.

tie commented:
@bogdandrutu since you’ve recently touched the logging exporter (#4399), perhaps you can take a look at this?

bogdandrutu commented on internal/otlpjson/databuffer.go:
```diff
@@ -0,0 +1,239 @@
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+type dataBuffer struct {
```
https://github.com/open-telemetry/opentelemetry-collector/blob/main/model/otlp/json_marshaler.go

bogdandrutu replied:
Already implemented, don't need to re-implement this.

tie replied:
Thanks, didn’t notice that a JSON serializer already exists. Unfortunately, the format is not “greppable” since a single line contains many spans (and that’s also the issue with `otlptext`, except that spans also take multiple lines). Maybe we can rename `otlpjson` to `loggingexporter/jsonformat` to indicate that it is a format used specifically for logging?

tie commented:
I’ve rebased the PR, added separate formats for `json` (standard) and `jsonstream` (“greppable”) marshalers, and moved `internal/otlpjson` to `exporter/loggingexporter/internal/jsonstream`.

jessequinn commented:
what is missing to get this merged? This is a desirable option for logging exporter.

jpkrohling commented:
I'll review this soon, but perhaps @bogdandrutu also wants to, given he did the first pass already?
